src/core/mac/wakeup_tx_scheduler.cpp

```diff
@@ -99,7 +99,7 @@ Mac::TxFrame *WakeupTxScheduler::PrepareWakeupFrame(Mac::TxFrames &aTxFrames)
     VerifyOrExit(frame->GenerateWakeupFrame(Get<Mac::Mac>().GetPanId(), target, source) == kErrorNone, frame = nullptr);
     frame->SetTxDelayBaseTime(0);
     frame->SetTxDelay(radioTxUs);
-    frame->SetCsmaCaEnabled(false);
+    frame->SetCsmaCaEnabled(kWakeupFrameTxCca);
```

Comment:
It seems like we are changing the behavior in this PR. It was previously set to `false`, but now it is configurable and defaults to `true` since we define `OPENTHREAD_CONFIG_WAKEUP_FRAME_TX_CCA_ENABLE` as `1`.

Could this have any consequences? @edmont, @Damian-Nordic, what do you think?

Reply:
**I think it would be good to clarify this (and have it documented in `radio.h`)**

I did a bit of checking in the `git history` for this.

- It was originally added in #2344.
  - It was used for the `OPENTHREAD_CONFIG_DISABLE_CCA_ON_LAST_ATTEMPT` feature.
  - The idea for this configuration was to always send on the last attempt (basically, do not perform CCA (carrier sense)).
- Later, this variable was renamed in #2844.
  - This was based on a discussion in #2801.
  - The above PR seems to change the interpretation to CSMA-CA and not the CCA mechanism.
  - It also renamed `DISABLE_CCA_ON_LAST_ATTEMPT` to `DISABLE_CSMA_CA_ON_LAST_ATTEMPT`.
- `OPENTHREAD_CONFIG_DISABLE_CSMA_CA_ON_LAST_ATTEMPT` was later removed in #4765.

--------

I also checked how it is used today in the code:

Within OT core
- It is set to `true` by default in `Mac::Links::TxFrames::Clear()`.
- `CslTxScheduler::HandleFrameRequest()` and `WakeupTxScheduler::PrepareWakeupFrame()` set it to `false` when preparing the frame.

Within NCP/RCP code
- In `NcpBase::DecodeStreamRawTxRequest()`, it is set to `true` by default.
- It is later updated if this field is provided in the encoded spinel frame received from RCP.
- The current version of RCP seems to encode and provide this field in `RadioSpinel::Transmit()`.

Reply:
I agree we should add some missing clarification on the behavior with some combination of parameters in TxInfo.

@Damian-Nordic here's the latest alignment https://github.com/openthread/openthread/issues/2801. With https://github.com/openthread/openthread/pull/11012, the backoff should be skipped if mMaxBackoffs is 0.

Reply:
@bukepo So to summarize:
- `mCsmaCaEnabled` meant to enable full CSMA-CA, not just CCA.
- This is probably why this flag was set to false for CSL (which was also inherited for wake-up).
- For Nordic's NCS platform setting this flag to true for scheduled TX doesn't change anything (our platform always does CCA for scheduled TX), though this seems like a change that might impact some other platforms.
- Given the above I'm fine with this changing the behavior so that `mCsmaCaEnabled && mMaxCsmaBackoffs == 0` requests a single CCA, but this needs to be better documented as it clearly confused people in the past :).

Reply:
I'll submit a PR to document the desired behavior.